src/lib/event.ts

```diff
@@ -178,5 +138,41 @@ export function event<
         return
     }
     Object.assign(handler, definition)
-    return handler as unknown as D & Handler<APIGatewayProxyEvent>
+    return handler as AWSLambdaHandler
+}
+
+export function firehose<C, S, FirehoseP, FirehoseR, D>(
+    definition: D & FirehoseTransformationHandler<C, S, FirehoseP, FirehoseR>
+): AWSLambdaHandler & D {
+    return event(definition as unknown as EventHandler) as AWSLambdaHandler & D
+}
+
+export function eventBridge<C, S, EbP, EbR, D>(definition: D & EventBridgeHandler<C, S, EbP, EbR>): AWSLambdaHandler & D {
+    return event(definition as unknown as EventHandler) as AWSLambdaHandler & D
+}
+
+export function http<C, S, HttpB, HttpP, HttpQ, HttpH, HttpR, GV extends GatewayVersion, D>(
+    definition: D & HttpHandler<C, S, HttpB, HttpP, HttpQ, HttpH, HttpR, GV>
+): AWSLambdaHandler & D {
+    return event(definition as unknown as EventHandler) as AWSLambdaHandler & D
+}
+
+export function kinesis<C, S, KinesisP, KinesisR, D>(
```

Comment:
I'm afraid these names are too ambiguous, and will clash with application code (same for `http`, `sns`, etc). what about `kinesisHandler` or `kinesisEventHandler`?